File: wallet.py
portfeuille = []
menu = 0

while menu != 5:
    print("Veuillez choisir une action")
    print("1) Ajouter un élément")
    print("2) Afficher les contenus du portefeuille")
    print("3) Retirer un élément du portefeuille")
    print("4) Vider le portefeuille")
    menu = int(input())

    if menu == 1:
        contenu = input("Veuillez ajouter un contenu : ")
        if len(contenu) != 0:
            portfeuille.append(contenu)
            print("Vous avez ajouté un contenu dans votre portefeuille")

    elif menu == 2:
        nbr = 1
        for contenu in portfeuille:
            print(nbr, contenu)
            nbr += 1

    elif menu == 3:
        print("Contenu du portefeuille :")
        nbr = 1
        for contenu in portfeuille:
            print(nbr, contenu)
            nbr += 1

        rm = int(input("Sélectionnez un élément à retirer : "))
        if 0 < rm < len(portfeuille):
            portfeuille.pop(rm - 1)
            print("Un élément a été retiré du portefeuille")

    elif menu == 4:
        portfeuille.clear()
        print("Le portefeuille a été vidé")

    elif menu == 5:
        print("Vous avez quitté le programme")

    else:
        print("Veuillez choisir le bon menu")

File: Wallet.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Wallet {
    public static void main(String[] args) {
        List<String> portfeuille = new ArrayList<>();
        int menu = 0;

        Scanner scanner = new Scanner(System.in);

        while (menu != 5) {
            System.out.println("Veuillez choisir une action");
            System.out.println("1) Ajouter un élément");
            System.out.println("2) Afficher les contenus du portefeuille");
            System.out.println("3) Retirer un élément du portefeuille");
            System.out.println("4) Vider le portefeuille");
            System.out.println("5) Quitter");

            menu = Integer.parseInt(scanner.nextLine());

            if (menu == 1) {
                System.out.println("Veuillez ajouter un contenu :");
                String contenu = scanner.nextLine();
                if (!contenu.isEmpty()) {
                    portfeuille.add(contenu);
                    System.out.println("Vous avez ajouté " + contenu + " dans votre portefeuille");
                }
            } else if (menu == 2) {
                int nbr = 1;
                for (String contenu : portfeuille) {
                    System.out.println(nbr + " " + contenu);
                    nbr++;
                }
            } else if (menu == 3) {
                System.out.println("Contenu du portefeuille :");
                int nbr = 1;
                for (String contenu : portfeuille) {
                    System.out.println(nbr + " " + contenu);
                    nbr++;
                }

                System.out.println("Sélectionnez un élément à retirer :");
                int rm = Integer.parseInt(scanner.nextLine());
                if (rm > 0 && rm <= portfeuille.size()) {
                    portfeuille.remove(rm - 1);
                    System.out.println("Un élément a été retiré du portefeuille");
                }
            } else if (menu == 4) {
                portfeuille.clear();
                System.out.println("Le portefeuille a été vidé");
            } else if (menu == 5) {
                System.out.println("Vous avez quitté le programme");
            } else {
                System.out.println("Veuillez choisir le bon menu");
            }
        }
    }
}
